plugins/modules/win_timezone.ps1

```diff
@@ -3,71 +3,75 @@
 # Copyright: (c) 2015, Phil Schwartz <schwartzmx@gmail.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
 
-$params = Parse-Args $args -supports_check_mode $true
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
-$diff_support = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+$spec = @{
+    options = @{
+        timezone = @{
+            required = $true
+            type = "str"
+        }
+    }
+    supports_check_mode = $true
+}
 
-$timezone = Get-AnsibleParam -obj $params -name "timezone" -type "str" -failifempty $true
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 
-$result = @{
-    changed = $false
-    previous_timezone = $timezone
-    timezone = $timezone
-}
+$timezone = $module.Params.timezone
```

Comment:
Instead of using `Parse-Args` and `Get-AnsibleParam` we define the argument spec inside a hashtable, create the module object, then retrieve the module options through `$module.Params.x`. The `$spec` is the module option specification and this module only has 1 option called `timezone` that is required and is typed as a string.

We also explicitly have to opt into saying we support check mode with `supports_check_mode = $true` in the root of the spec hashtable. Without this the module will be skipped when running under check mode.